### PR TITLE
[Dialogs] Allow setting text alignment for alert message text

### LINKED ISSUE #5602: [Dialogs] Allow setting text alignment for alert message text
This was filed as an internal issue. If you are a Googler, please visit [b/118846596](http://b/118846596) for more details.

---
#### Internal data
- Associated internal bug: [b/118846596](http://b/118846596)

### PR DESCRIPTION
**Pull request by galiak11**
# Description

Using NSTextAlignment to customize alignment of the Material Alert's message.

## Example
Trailing aligned message with a leading aligned message:

Dialogs - message alignment

## Issue
Closes #5602 

b/118846596, cl/298994401

### CODE REVIEW

**Comment by jverkoey** [comment]
Please associate the PR with the GitHub issue for tracking purposes by putting something like:

`Closes <link to GitHub Issue>`

in the PR description.

File: components/Dialogs/src/MDCAlertController.m [hunk 1]
```diff
@@ -329,6 +330,14 @@ - (void)setTitleAlignment:(NSTextAlignment)titleAlignment {
   }
 }
 
+// todo: test alignment getter - after changing the alert.view.messageAlignment
```
Comment (jverkoey) [inline components/Dialogs/src/MDCAlertController.m]: Please file an Issue to track this TODO and link to it here in the form `TODO(<link to Issue>): ...`.

**Review by jverkoey — COMMENTED** [review]
> Trailing aligned message with a leading aligned message:

Do you mean to say "An example with a trailing-aligned message" ?

File: components/Dialogs/src/MDCAlertController.m (hunk 1 above)
Comment (galiak11) [reply components/Dialogs/src/MDCAlertController.m]: Removed the comment.

I tested this, and it seems all Dialogs getters return incorrect values after a different value is set directly on the view. For example:
```
MDCAlertController *alertview = (MDCAlertController *)alert.view;
alertview.<getter> = <value>; 
print (alert.<getter>);   // prints original value before the change.
```
To remain consistent with other getters, I repeated this behavior in the new getter.

**Review by jverkoey — APPROVED** [review]
LGTM, but please add a snapshot test for the new API prior to landing this PR.

File: components/Dialogs/src/MDCAlertController.h [hunk 2]
```diff
@@ -98,27 +98,30 @@
...
-/** The tint color applied to the titleIcon. Leave empty to preserve original image color(s).*/
+/** The tint color applied to the titleIcon. Leave empty to preserve original image color(s). */
 @property(nonatomic, strong, nullable) UIColor *titleIconTintColor;
 
-/** The font applied to the message of Alert Controller.*/
+/** The font applied to the message of the Alert Controller. */
 @property(nonatomic, strong, nullable) UIFont *messageFont;
 
-/** The color applied to the message of Alert Controller.*/
+/** The color applied to the message of the Alert Controller. */
```
Comment (jverkoey) [inline components/Dialogs/src/MDCAlertController.h]: Please move the unrelated text changes here out to a separate cleanup PR.

**Review by jverkoey — COMMENTED** [review]
testMessageRightAlignmentIsRightInRTL_11_2@2x.png looks like it may have uncovered a bug. Please file an issue for it accordingly.

**Comment by galiak11** [comment]
These text changes were added in response to review feedback in corresponding CL.

**Comment by jverkoey** [comment]
This was rolled back in https://github.com/material-components/material-components-ios/pull/9905

https://github.com/material-components/material-components-ios/issues/9909 to track rolling forward.

**Comment by jverkoey** [comment]
Rolled forward in the release. The internal testing issue appears to be due to timeouts and not these changes.
